packages/bpk-component-button/src/BpkButtonDestructive.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import React from 'react';
+
+import { type Props, defaultProps, propTypes } from './common-types';
+import STYLES from './BpkButtonDestructive.scss';
+
+// This was originally depended upon from the bpk-react-utils package, however
+// we decided to inline it in this particular component so as not to bloat the
+// the bundles of consumers who are not yet on webpack 2
+// We'll revisit this again soon.
+const cssModules = (styles = {}) => className =>
+  styles[className] ? styles[className] : className;
+
+const getClassName = cssModules(STYLES);
+
+const BpkButtonDestructive = (props: Props) => {
+  const {
+    children,
+    href,
+    className,
+    onClick,
+    disabled,
+    submit,
+    large,
+    iconOnly,
+    blank,
+    rel: propRel,
+    ...rest
+  } = props;
+
+  const classNames = [getClassName('bpk-button--destructive')];
```

Comment:
Can we avoid the duplication of code here and in the all other types? I think ideally we should have `BpkButtonBase` component and render it here with the appropriate props/class name

Reply:
I agree, but would prefer a different method for code sharing. Composition over inheritance

Reply:
That's what I meant. Something like rendering `<BpkButtonBase />` inside this function, not extending from it

Reply:
I have pushed a commit that introduces a `BpkButtonBase` class that reduces the amount of repeated code a lot. It's much better now 😄